src/app/shared/components/product/product-variation-select/product-variation-select.component.ts

```diff
@@ -36,10 +38,22 @@ export class ProductVariationSelectComponent implements OnChanges, OnDestroy {
   initForm() {
     if (this.variationOptions) {
       this.form = this.buildSelectForm(this.variationOptions);
-      this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(this.selectVariation);
+      this.initialSelection = this.form.getRawValue();
+      this.form.valueChanges.pipe(takeUntil(this.destroy$)).subscribe(selection => {
+        this.selectVariation.emit({ selection, changedAttribute: this.getChangedAttribute(selection) });
+      });
     }
   }
 
+  getChangedAttribute(selection) {
+    const diff = reduce(
+      selection,
+      (result, value, key) => (isEqual(value, this.initialSelection[key]) ? result : result.concat(key)),
+      []
+    );
```

Comment:
What is the advantage of using `reduce` from `lodash` instead of the built-in `selection.reduce`?